sonar-plugin-api/src/main/java/org/sonar/api/ce/posttask/PostProjectAnalysisTaskTester.java

```diff
@@ -177,60 +179,41 @@ public PostProjectAnalysisTaskTester withBranch(@Nullable Branch b) {
     return this;
   }
 
-  public void execute() {
+  /**
+   * @since 6.6
+   */
+  public PostProjectAnalysisTaskTester withAnalysisUuid(@Nullable String analysisUuid) {
+    this.analysisUuid = analysisUuid;
+    return this;
+  }
+
+  public PostProjectAnalysisTask.ProjectAnalysis execute() {
     requireNonNull(ceTask, CE_TASK_CAN_NOT_BE_NULL);
     requireNonNull(project, PROJECT_CAN_NOT_BE_NULL);
     requireNonNull(date, DATE_CAN_NOT_BE_NULL);
 
-    this.underTest.finished(
-      new PostProjectAnalysisTask.ProjectAnalysis() {
-        @Override
-        public ScannerContext getScannerContext() {
-          return scannerContext;
-        }
-
-        @Override
-        public CeTask getCeTask() {
-          return ceTask;
-        }
-
-        @Override
-        public Project getProject() {
-          return project;
-        }
-
-        @Override
-        public Optional<Branch> getBranch() {
-          return Optional.ofNullable(branch);
-        }
-
-        @Override
-        public QualityGate getQualityGate() {
-          return qualityGate;
-        }
+    Analysis analysis = null;
+    if (date != null && analysisUuid != null) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2589)